packages/react-chat/src/contexts/RuntimeContext/index.tsx

```diff
@@ -4,19 +4,21 @@ import { RuntimeState, Settings, useRuntimeState } from './useRuntimeState';
 
 // split up API and state to prevent unnecessary re-renders
 export const RuntimeStateAPIContext = createContext<RuntimeState['api']>({} as any);
-export const RuntimeStateContext = createContext<RuntimeState['state']>({} as any);
+export const RuntimeStateContext = createContext<RuntimeState['state'] & { shadowRoot: ShadowRoot; autostart: boolean }>({} as any);
 
-interface RuntimeProviderProps extends React.PropsWithChildren, Settings {}
+interface RuntimeProviderProps extends React.PropsWithChildren, Settings {
+  shadowRoot: ShadowRoot;
+}
 
-export const RuntimeProvider = ({ children, assistant, config }: RuntimeProviderProps) => {
+export const RuntimeProvider = ({ children, assistant, config, shadowRoot }: RuntimeProviderProps) => {
   const store = useRuntimeState({ assistant, config });
 
   // api is a static object, so we can use useMemo to prevent unnecessary re-renders
   const api = useMemo(() => store.api, []);
 
   return (
     <RuntimeStateAPIContext.Provider value={api}>
-      <RuntimeStateContext.Provider value={store.state}>{children}</RuntimeStateContext.Provider>
+      <RuntimeStateContext.Provider value={{ ...store.state, shadowRoot, autostart: !!config.autostart }}>{children}</RuntimeStateContext.Provider>
```

Comment:
I think we need to memoize this value first otherwise we can cause unnecessary renders